### DISCUSSION TITLE
maraDNS as a simple relay / rewrite server

### DISCUSSION THREAD

**Discussion by dschuermans** [samboy/MaraDNS#106] [Q&A] [answered]
Hey,

I'm looking for a small, lightweight DNS server that's capable of simply relaying DNS queries.
I'd like to configure each computer on the network to use the server running maraDNS as their DNS server and for 1 particular domain, I'd like to return an internal IP (192.168.*.*) instead of the public ip.

Is such a use-case achievable with maraDNS?

It's all windows systems here and at first I thought I'd go with modifying the HOSTS file on each computer, but some of them are laptops which are taken elsewhere and if they request that one particular domain, I do want it to return the public IP if they're not connected to our LAN

**Comment by samboy** [comment, marked as answer]
For this, you use the `Deadwood`, not MaraDNS daemon.

Here’s an example config file. A more full example is here: https://github.com/samboy/MaraDNS/blob/master/deadwood-github/doc/dwood3rc-all

```
# IP Deadwood runs as
bind_address="127.0.0.1" 

# Directory we run program from (not used in Win32)
chroot_dir = "/etc/deadwood" 

# By default, for security reasons, Deadwood does not allow IPs in the
# 192.168.x.x, 172.[16-31].x.x, 10.x.x.x, 127.x.x.x, 169.254.x.x,
# 224.x.x.x, or 0.0.x.x range.  If using Deadwood to resolve names
# on an internal network, set this to 0
filter_rfc1918 = 0

# Who is allowed to use the cache
recursive_acl =  "127.0.0.1/16"
recursive_acl += ", 10.1.0.0/16" 
recursive_acl += ", 192.168.1.0/24" 

# Servers we connect to that themselves recursively solve DNS records
# Please note that each upstream_servers entry takes up space in Deadwood's
# cache and that maximum_cache_elements will need to be increased to store
# a large number of these entries.
upstream_servers = {}
upstream_servers["."]="8.8.8.8, 8.8.4.4"  # Google DNS

# We can have given names resolve to bogus IPv4 addresses.  Here, we
# have the name "maradns.foo" resolve to "10.10.10.10" and "kabah.foo"
# resolve to "10.11.11.11", regardless of what real values these DNS
# records may have.  Note that a given name can only resolve to a
# single IP, and that the records have a fixed TTL of 30 seconds.
ip4 = {}
ip4["maradns.foo."] = "10.10.10.10"
ip4["kabah.foo."] = "10.11.11.11"
```

**Comment by samboy** [comment]
Locking discussion to stop spam bots.
